version: "2.1"

services:
  app:
    build: .
    ports:
      - "8000:8000"
    links:
      - pg
    depends_on:
      pg:
        condition: service_healthy
    environment:
      - POSTGRES_HOST=pg
      - POSTGRES_PORT=5432
      - POSTGRES_DB=larek_db
      - POSTGRES_USER=larek_user
      - POSTGRES_PASSWORD=larek_password
      - DEBUG=true
    command: bash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
     - .:/app/

  pg:
    image: postgres:12
    environment:
      - POSTGRES_DB=larek_db
      - POSTGRES_USER=larek_user
      - POSTGRES_PASSWORD=larek_password
      - PGDATA=/var/lib/postgresql/data/pgdata
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U larek_user -d larek_db" ]
      interval: 5s
      timeout: 5s
      retries: 5

  pgadmin:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: root
    ports:
      - "5050:80"
